chat/anthropic_chat.py

```diff
@@ -0,0 +1,44 @@
+from anthropic import Anthropic
+from .engine import ChatEngine
+from chat.exceptions import ChatResponseGenerationError
+
+
+class ChatAnthropic(ChatEngine):
+
+    def __init__(self):
+        self.client = Anthropic(api_key="aaaaa")
+        self.history = []
+
+    def generate_response(self, query: str, context: str = None) -> str:
+        if not query:
+            return ""
+
+        if context:
+            full_input = f"Given a context: {context}\n Given a query: {query}\n Please answer query based on the given context."
+        else:
+            full_input = query
+
+        self.history.append({"role": "user", "content": full_input})
+        
+        try:
+            response = self.client.messages.create(
+                model="claude-3-5-sonnet-20240620",
+                max_tokens = 1024,
+                messages=self.history,
+                system="""
+                        You are an assistant that can only provide responses based on the PROVIDED CONTEXT. 
+                        DO NOT USE AN EXTERNAL/GENERAL KNOWLEDGE, only answer based on PROVIDED CONTEXT. !!!
+                        """
+            )
```

Comment:
Any explanations on why we use this certain model, max tokens, and system prompt?

Reply:
Hey @arkanalexei, referring to this [API Reference by Anthropic](https://www.anthropic.com/pricing#anthropic-api), it seems that the most cost-effective model is actually Claude 3 Haiku. I think for this PR I will change it to Haiku instead of Sonnet to save on money.

Regarding the max tokens and system prompt, it's completely arbitrary as of now.